aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1227,13 +1227,15 @@ Tensor& bmm_out_cpu(const Tensor& batch1, const Tensor& batch2, Tensor &result)
 }
 
 Tensor& dot_out(const Tensor& self, const Tensor& tensor, Tensor& result) {
+  checkSameDevice("dot_out", result, self);
```

Comment:
@anjali411, here, I'm only checking if `self` and `result` are on the same device, but not `tensor`.
Please advise if I should add the following line as well:
`checkSameDevice("dot_out", result, tensor);`

There's a function `checkAllSameGPU()` for checking if any number of tensors are on the same GPU,
but no such function exists for all devices (such as `checkAllSameDevice`).

BTW, the CI failure is unrelated.
`test_out` passed for both `dot` & `vdot`.

Reply:
Looks like using `checkSameDevice("dot", result, tensor);` results in a weird build failure

Reply:
```
/var/lib/jenkins/workspace/aten/src/ATen/native/LinearAlgebra.cpp:1240:3: error: no matching function for call to 'checkSameDevice'
 checkSameDevice("vdot", result, tensor);
 ../../aten/src/ATen/native/LinearAlgebraUtils.h:485:20: note: candidate function not viable: no overload of 'tensor' matching 'at::Tensor' for 3rd argument
```

Reply:
I simply renamed `tensor` to `other`